sunpy/net/tests/test_helioviewer.py

```diff
@@ -88,16 +90,20 @@ def test_get_jp2_header(self, client):
         assert ('fits' in header1.keys()) and ('fits' in header2.keys())
 
     @skip_glymur
+    @figure_test
     def test_download_jp2_map(self, client):
         """
-        Tests getJP2Image API method with Map.
+        Tests getJP2Image API method with Map with a figure test.
         """
-        # TODO: make this a figure test.
-        filepath = client.download_jp2('2012/01/01', observatory='SOHO',
-                                       instrument='MDI', measurement='continuum')
-        map_ = sunpy.map.Map(filepath)
-        assert isinstance(map_, sunpy.map.GenericMap)
-        os.remove(filepath)
+        filepath = tempfile.TemporaryDirectory()
+        client.download_jp2('2012/01/01', observatory='SOHO',
+                            instrument='MDI', measurement='continuum', directory=tempfile.name)
```

Comment:
Could we not use TemporaryDirectory as a context manager? 

In fact we could use pytests tmpdir instead and that should auto clean up at the end of the test.

Reply:
Okay, shall I make this change?

Reply:
Sure.